wagtail_wordpress_import/block_builder_defaults.py

```diff
@@ -327,8 +328,18 @@ def document_exists(name):
 
 def get_absolute_src(src, domain_prefix=None):
     src = src.lstrip("/")
-    if not src.startswith("http") and domain_prefix:
-        return domain_prefix + "/" + src
+    url = urlparse(src)
+    known_file_extensions = ["jpg", "jpeg", "gif", "png", "webp"]
+
+    if not url.scheme and domain_prefix:
+        first_url_part = url.path.split("/")[0]
+        if (
+            "." in first_url_part
+            and first_url_part.split(".")[1].lower() not in known_file_extensions
```

Comment:
**Issue**: In the following case, the unrecognised local file extension is interpreted as a domain:

```python
>>> get_absolute_src("image.bmp", "https://www.example.com")
'https://image.bmp'
```

`urlparse` results have a `.hostname` attribute that you can use in this situation. You will have to remove the `.lstrip("/")` line though.

```python
>>> from urllib.parse import urlparse
>>> urlparse("image.bmp").hostname
>>> urlparse("example.com/image.bmp").hostname
>>> urlparse("//example.com/image.bmp").hostname
'example.com'
```